### PR TITLE
Added tracing

### PR DESCRIPTION
**Pull request by Jakobzs**
This PR adds `tracing` as a logger over `log` and `simplelog`, along with a util function for showing color in the console.

Logging is kind of a personal preference in terms of which library to use, but some reasons for choosing `tracing` is that it's hugely more popular, featureful in comparison to the other solutions, and is also managed by the Tokio team. It also helps that the current logging implementation barely requires any changes to adapt to `tracing` 😄

### CODE REVIEW

**Comment by Godnoken** [comment]
Out of curiosity, what is the purpose of this change?

File: tests/dx11.rs [hunk 1]
```diff
@@ -54,9 +53,6 @@ fn test_imgui_dx11() {
         }
     }
 
-    TermLogger::init(LevelFilter::Trace, Config::default(), TerminalMode::Mixed, ColorChoice::Auto)
-        .unwrap();
-
```
Comment (veeenu) [inline tests/dx11.rs:59]: Can we add a tracing subscriber here? Seeing logs is pretty important for tests (same for other tests).

File: hudbook/hello-hud/Cargo.toml [hunk 2]
```diff
@@ -14,5 +14,5 @@ crate-type = ["cdylib", "rlib"]
 name = "hello_hud"
 
 [dependencies]
-hudhook = { version = "0.4.0", path = "../..", features = ["simplelog"] }
+hudhook = { version = "0.4.0", path = "../..", features = [] }
```
Comment (veeenu) [inline hudbook/hello-hud/Cargo.toml:17]:
```suggestion
hudhook = { version = "0.4.0", path = "../.." }
```

File: examples/dx11_hook.rs [hunk 3]
```diff
@@ -9,8 +9,6 @@ impl Dx11HookExample {
     fn new() -> Self {
         println!("Initializing");
         hudhook::utils::alloc_console();
-        #[cfg(feature = "simplelog")]
-        hudhook::utils::simplelog();
```
Comment (veeenu) [inline examples/dx11_hook.rs:13]: Same as for the tests below -- can we add a tracing subscriber here?

File: Cargo.toml [hunk 4]
```diff
@@ -99,17 +98,16 @@ windows = { version = "0.39.0", features = [
   "Win32_UI_Input_KeyboardAndMouse",
   "Win32_UI_WindowsAndMessaging",
 ] }
-simplelog = { version = "0.12.0", optional = true }
+tracing = "0.1"
 
 [dev-dependencies]
-simplelog = "0.12.0"
+# tracing-subscriber = "0.3"
```
Comment (veeenu) [inline Cargo.toml:104]: I'd remove the `dev-dependencies` table entirely if we don't use any of its values; but we may need this anyway (see other comments)

**Review by veeenu — CHANGES_REQUESTED** [review]
Thank you for this! From a cursory check, it looks good in general, but I'd add logging in test/example modules. Haven't had a chance to test it but will do so soon.

**Comment by Jakobzs** [comment]
Tracing subscribers should be added now.

**Review by veeenu — CHANGES_REQUESTED** [review]
I'm a bit scared because it took me quite a while to adapt one practice tool's code to work with tracing, so this will be quite the breaking change. But it seems like it also improved code quality somewhat and I prefer `tracing`'s approach. Once the two modifications I suggested are ready, we can merge.

File: src/lib.rs [hunk 5]
```diff
@@ -232,12 +239,13 @@ pub mod lifecycle {
     }
 }
```
[the remark this answers is not in the document]
Comment (Sages0ft) [reply src/lib.rs:241]: Done

**Review by veeenu — APPROVED** [review]

**Comment by Godnoken** [comment]
> It is mainly to integrate a hugely more popular logging framework into hudhook.

Sorry for not getting back to this. Been crunching on my own stuff for a bit.

Interesting. Without having had a look at it - I assume I will no longer be able to see logs from hudhook from my own application if I stick to log and simplelog? I just implemented it a while ago and I'm quite content with it, but if not I'll have to change.

**Comment by veeenu** [comment]
> Interesting. Without having had a look at it - I assume I will no longer be able to see logs from hudhook from my own application if I stick to log and simplelog? I just implemented it a while ago and I'm quite content with it, but if not I'll have to change.

Yep, that's correct. It's quite a big breaking change, but luckily the user facing API is almost 1:1 with tracing. Check [this PR](https://github.com/veeenu/eldenring-practice-tool/pull/46/files) out -- it looks big because it has other unrelated lints, but the change should be pretty small.

**Comment by Godnoken** [comment]
Fair enough, I'll take a look. Cheers!

**Pull request merged**